plugins/inputs/redis_sentinel/redis_sentinel.go

```diff
@@ -0,0 +1,609 @@
+			if val, err := strconv.ParseInt(val, 10, 64); err == nil {
+				fields[key] = val
+			} else {
+				acc.AddError(err)
+			}
+		case "string":
+			fields[key] = val
+		default:
+			continue
+		}
+	}
+
+	return tags, fields
+}
+
+// converts `sentinel sentinels <name>` output to tags and fields
+func convertSentinelSentinelsOutput(
+	acc telegraf.Accumulator,
+	globalTags map[string]string,
+	masterName string,
+	sentinelMaster map[string]string,
+) (map[string]string, map[string]interface{}) {
+	tags := make(map[string]string)
+	for k, v := range globalTags {
+		tags[k] = v
+	}
+
+	tags["sentinel_ip"] = sentinelMaster["ip"]
+	tags["sentinel_port"] = sentinelMaster["port"]
+	tags["master"] = masterName
+
+	fields := make(map[string]interface{})
+
+	for key, val := range sentinelMaster {
+		key = strings.Replace(key, "-", "_", -1)
+
+		switch valType := measurementSentinelsFields[key]; valType {
+		case "float":
+			if val, err := strconv.ParseFloat(val, 64); err == nil {
+				fields[key] = val
+			} else {
+				acc.AddError(err)
+			}
+		case "integer":
+			if val, err := strconv.ParseInt(val, 10, 64); err == nil {
+				fields[key] = val
+			} else {
+				acc.AddError(err)
+			}
+		case "string":
+			fields[key] = val
+		default:
+			continue
+		}
+	}
+
+	return tags, fields
+}
+
+// converts `sentinel replicas <name>` output to tags and fields
+func convertSentinelReplicaOutput(
+	acc telegraf.Accumulator,
+	globalTags map[string]string,
+	masterName string,
+	replica map[string]string,
+) (map[string]string, map[string]interface{}) {
+	tags := make(map[string]string)
+	for k, v := range globalTags {
+		tags[k] = v
+	}
+
+	tags["replica_ip"] = replica["ip"]
+	tags["replica_port"] = replica["port"]
+	tags["master"] = masterName
+
+	fields := make(map[string]interface{})
+
+	for key, val := range replica {
+		key = strings.Replace(key, "-", "_", -1)
+
+		switch valType := measurementReplicasFields[key]; valType {
+		case "float":
+			if val, err := strconv.ParseFloat(val, 64); err == nil {
+				fields[key] = val
+			} else {
+				acc.AddError(err)
+			}
+		case "integer":
+			if val, err := strconv.ParseInt(val, 10, 64); err == nil {
+				fields[key] = val
+			} else {
+				acc.AddError(err)
+			}
+		case "string":
+			fields[key] = val
+		default:
+			continue
+		}
+	}
+
+	return tags, fields
+}
+
+// convertSentinelInfoOutput parses `INFO` command output
+// Largely copied from the Redis input plugin's gatherInfoOutput()
+func convertSentinelInfoOutput(
+	acc telegraf.Accumulator,
+	globalTags map[string]string,
+	rdr io.Reader,
+) (map[string]string, map[string]interface{}) {
+	scanner := bufio.NewScanner(rdr)
+	fields := make(map[string]interface{})
+
+	tags := make(map[string]string)
+	for k, v := range globalTags {
+		tags[k] = v
+	}
+
+	var section string
+	for scanner.Scan() {
+		line := scanner.Text()
+		if len(line) == 0 {
+			continue
+		}
+
+		if line[0] == '#' {
+			if len(line) > 2 {
+				section = line[2:]
+			}
+			continue
+		}
```

Comment:
Can you please add a comment here on what the reasoning behind this code is.

Reply:
@srebhan Addressed in https://github.com/influxdata/telegraf/pull/10042/commits/bfe4ae4463f16cbd0d7db55a70b6c0cc53c6a5da

Reply:
I did not yet get why you store the section if the line is longer than two? Why not one?

Reply:
@srebhan I refactored that entire section of the code in https://github.com/influxdata/telegraf/pull/10042/commits/e6e8cb688147ec560dd7c45fa0cf063c6ab3fe4a, so the comment won't be needed anymore I hope.

Reply:
I really like it.